### PR TITLE
Intersection fields

### PR DESCRIPTION
**Pull request by korshaknn**
added fields
`isUrban`
`Street`

### CODE REVIEW

**Comment by korshaknn** [comment]
@LukasPaczos, @Guardiola31337 doc's is the most difficult part) need your help)

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java [hunk 1]
```diff
@@ -146,6 +147,28 @@ public Point location() {
...
+  /**
+   * An object containing detailed information about the road exiting the intersection along the
+   * route. Properties in this object correspond to properties in the {@link #classes()}
+   * specification. Only available on the {@link DirectionsCriteria#PROFILE_DRIVING} profile.
+   *
+   * @return an object containing detailed road information.
+   */
+  @Nullable
+  @SerializedName("mapbox_streets_v8")
+  public abstract Street streets();
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:170]:
```suggestion
  public abstract MapboxStreetsV8 mapboxStreetsV8();
```

This value corresponds to streets v8 sources, so I think we need to be explicit here, even if it looks ugly.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Street.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
+import com.mapbox.api.directions.v5.DirectionsCriteria;
+
+/**
+ * An object containing detailed information about the road exiting the intersection along the
+ * route.
+ * Only available on the {@link DirectionsCriteria#PROFILE_DRIVING} profile.
+ */
+@AutoValue
+public abstract class Street extends DirectionsJsonObject {
```
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Street.java:18]: Same as above.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java (hunk 1 above)
Comment (korshaknn) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java:170]: fixed

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Street.java (hunk 2 above)
Comment (korshaknn) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Street.java:18]: fixed

**Review by LukasPaczos — APPROVED** [review]
:ship:

**Pull request merged**